src/bot_methods.js

```diff
@@ -347,11 +348,32 @@ function handleChallengeMode(message) {
       }
     }
 
+    definitionFrags = engWord.split(", "); // array contatining all possible definition
+
     console.log("challenge mode ran: " + estWord + ": " + engWord);
 
-    const filter = response => response.content.toLowerCase() === engWord.toLowerCase() // the filter for accepting user input.
-      || response.content.toLowerCase() === STOP  // if the user types stop, next or the correct word,
-      || response.content.toLowerCase() === NEXT; // it will return true. otherwise false
+    const filter = function (response) { // the filter function, which resolves the promise in the later message collector

```

Comment:
nice attempt, but correct me if i'm wrong - you can do something much simpler:

use the string [inclues() ](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/includes)function to check if it contains the user input

something like this (psuedocode):
userInput = input.toLowerCase()
translation = english.toLowerCase()
return translation.includes(userInput)